dbms/src/Flash/Mpp/MPPTunnelSet.h

```diff
@@ -50,11 +54,19 @@ class MPPTunnelSetBase : private boost::noncopyable
     // this is a partition writing.
     void write(tipb::SelectResponse & response, int16_t partition_id);
     void write(mpp::MPPDataPacket & packet, int16_t partition_id);
+    void writeError(const String & msg);
+    void close(const String & reason);
+    void finishWrite();
+    TunnelPtr getTunnelById(const MPPTaskId & id);
 
     uint16_t getPartitionNum() const { return tunnels.size(); }
 
-    void addTunnel(const TunnelPtr & tunnel)
+    void registerTunnel(const MPPTaskId & id, const TunnelPtr & tunnel)
```

Comment:
how about hide the implementation in .cpp file?

Reply:
Done